tests/repositories/test_legacy_repository.py

```diff
@@ -43,3 +44,11 @@ def test_page_absolute_links_path_are_correct():
     for link in page.links:
         assert link.netloc == "files.pythonhosted.org"
         assert link.path.startswith("/packages/")
+
+
+@mock.patch("poetry.repositories.legacy_repository.get_http_basic_auth")
+def test_http_basic_auth_repo(mock_get_auth):
```

Comment:
Testing this will be more complex with the changes I requested since we will need to check the url that will be called in `_get()`.